internal/tiltfile/k8s.go

```diff
@@ -382,43 +382,52 @@ func (s *tiltfileState) k8sResourceV2(thread *starlark.Thread, fn *starlark.Buil
 		return nil, err
 	}
 
+	resourceName := workload
+	needsToHaveObjects := false
 	if workload == "" {
-		return nil, fmt.Errorf("%s: workload must not be empty", fn.Name())
+		resourceName = newName
```

Comment:
(no changes needed, just grumbling) i feel a little weird about the way this way of doing things splits up the re/naming logic--some of it is handled here if workload is empty, some happens down on ln 386, and i think it wouldn't be obvious to someone new coming into this codebase where name assignment happens. i'd rather see `k8sResourceOptions` store in an array, and in assembly we iterate over them all and name or rename them in the same place.

of course that involves a different strategy for determining whether you've called `k8s_resource` for a given resource name, and i get it if you don't wanna bother with that refactor right now. figured i'd mention if you feel moved to change, but this isn't a big deal.